Makefile

```diff
@@ -191,6 +191,14 @@ SAIL_FLAGS += -c_coverage $(ALL_BRANCHES) -c_include sail_coverage.h
 C_LIBS += $(SAIL_LIB_DIR)/coverage/libsail_coverage.a -lm -lpthread -ldl
 endif
 
+# Optional link C_LIBS statically. Unlike -static this will not
+# link glibc statically which is generally a bad idea.
+ifneq (,$(STATIC))
+    C_LIBS_WRAPPED = -Wl,-Bstatic $(C_LIBS) -Wl,-Bdynamic
```

Comment:
Maybe worth noting this isn't a thing that works on macOS (nor does it have an equivalent beyond providing paths to the .a files or messing with the search paths)? Also `-Wl,--push-state,-Bstatic ... -Wl,--pop-state` is probably better practice, though in reality you're unlikely to be in an environment where it's not -Bdynamic.

Reply:
Ah very good point! I kind of totally forgot about MacOS existing! I will try and make this work there too. Mercifully `gmp` in Brew provides the static version.

And thanks for telling me about `--push-state`, `--pop-state`; I didn't know about that and it is definitely better.

Reply:
Updated with `--push/pop-state`, and also Mac support. Make and Mac's linker don't make it easy unfortunately, but it works.